lib/azure/armrest/insights/event_service.rb

```diff
@@ -44,22 +44,16 @@ def initialize(armrest_configuration, options = {})
         #   filter = "eventTimestamp ge #{date} and eventChannels eq 'Admin, Operation'"
         #   select = "resourceGroupName, operationName"
         #
-        #   ies.list(:filter => filter, :select => select, :all => true).events.each{ |event|
+        #   ies.list(:filter => filter, :select => select, :all => true).each{ |event|
         #     p event
         #   }
         #
         def list(options = {})
           url = build_url(options)
           response = rest_get(url)
-          json_response = JSON.parse(response.body)
 
-          events = ArmrestCollection.new(
-            json_response['value'].map do |hash|
-              Azure::Armrest::Insights::Event.new(hash)
-            end
-          )
-
-          events.continuation_token = parse_skip_token(json_response)
+          klass  = Azure::Armrest::Insights::Event
```

Comment:
That doesn't deal with skip tokens (which are in the body), and doesn't provide a lot of added value IMO. My approach handles skip tokens and reduces a lot of code.